hyperion/grid/voronoi_grid.py

```diff
@@ -450,10 +456,17 @@ def write(self, group, quantities='all', copy=True, absolute_paths=False, compre
         voronoi_table['volume'][np.isnan(voronoi_table['volume'])] = -1.
         voronoi_table['volume'][np.isinf(voronoi_table['volume'])] = -1.
 
-        if voronoi_table['neighbours'].dtype != np.int32:
-            raise TypeError("neighbours should be int32")
+        # Remove the dense neighbours from the table.
+        voronoi_table = voronoi_table['coordinates', 'volume', 'bb_min', 'bb_max']
```

Comment:
Just out of curiosity - would it be more efficient for the C++ code to simply return the sparse representation in the first place and not bother with the original representation? At the moment, the original table will still exist while a model is being set up, which will use more memory than is needed. Basically what is lost if you simply don't ever compute the ``neighbours`` table in the first place?